tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -139,19 +134,16 @@ public static OpenSpan startSpan(String operation) {
     }
 
     private static OpenSpan startSpanInternal(String operation, SpanType type) {
-        OpenSpan.Builder spanBuilder = OpenSpan.builder()
-                .operation(operation)
-                .spanId(Tracers.randomId())
-                .type(type);
-
         Trace trace = getOrCreateCurrentTrace();
         Optional<OpenSpan> prevState = trace.top();
+        final OpenSpan span;
         // Avoid lambda allocation in hot paths
         if (prevState.isPresent()) {
-            spanBuilder.parentSpanId(prevState.get().getSpanId());
+            span = OpenSpan.of(operation, Tracers.randomId(), type, Optional.of(prevState.get().getSpanId()));
```

Comment:
I realize this is existing and we're stuck with this API, but the `Optional` allocations on hot paths make me sad.

Reply:
me sad too, do you think it's a good idea to create three methods to avoid the `Optional` wrapping? 
```
static OpenSpan.of( .......other params...... Optional<String> parentSpanId) ;
static OpenSpan.of( .......other params......String parentSpanId) ;
static OpenSpan.of( .......other params.....) ;

```

Reply:
I don't think its worth it as they'll still get created

Reply:
👍